### PR TITLE
Addition of parser exchange options.

### PR DESCRIPTION
**Pull request by Darkoberd00**
Parsers can be exchanged via categories. 
Each parser has the same interface.

With the code: 
```
import { parser } from “../../utils/parser”.
export default parser(“ParserName”, ({content, log}) => {
do //parser things
return “any string
})
```

creates a parser.
The name of the parser must be specified in lower case.

The parser name is saved via the channel ID and the correct parser is automatically loaded when writing to the channel.

```
const parserDB = await prisma.parserdle.findFirst({
		where: { channelID: interaction.channel.id })
	})
....

const parser = (allParsersMap.get(parserDB.parserdleName))
```

### CODE REVIEW

**Comment by Schmarsch** [comment]
gonna have to try this out later on.
I only read over the code just now but it looks good to me currently.

Before I merge this I first want to add some actions to the repository.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -19,6 +19,12 @@ model User {
   LoldleDaily LoldleDaily[]
 }
 
+model Parserdle {
+  channelID String @unique @id
+  guildID String
+  parserdleName String
+}
+
 model LoldleDaily {
```
Comment (Schmarsch) [inline prisma/schema.prisma]: please rename this model because we now can suppport multiple dles

File: src/commands/general/setParser.ts
Comment (Schmarsch) [inline src/commands/general/setParser.ts:1]: This works for now, at a later stage I would like to extract this functionality into a web page for better design and readability.

Maybe move this into debug for now.

File: src/parser/index.ts
Comment (Schmarsch) [inline src/parser/index.ts:1]: Please add Documentation explaining the used Pattern

File: src/commands/general/setParser.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    async ({ interaction }) => {
+        await interaction.deferReply({ ephemeral: true });
+        const parser = interaction.options.getString("parser");
+        const channelid = interaction.channel?.id;
+        const guildid = interaction.guild?.id;
+
+        if (!channelid || !guildid) return interaction.editReply("Not in a guild channel");
+        if (!parser) return interaction.editReply("No parser provided");
+
+        const parserDB = await prisma.parserdle.upsert({
```
Comment (Schmarsch) [inline src/commands/general/setParser.ts]: please check if the selected parser is already the set parser, if so print a warning message.

If possible give the user the ability to change it in that message.

**Review by Schmarsch — CHANGES_REQUESTED** [review]
Please fix all requested changes.

Also please add Documentation if don't start now we never will!

File: src/commands/general/setParser.ts (hunk 2 above)
Comment (Darkoberd00) [reply src/commands/general/setParser.ts]: Of course i can do it, but it would take longer. i would build it similar to the help comand.

Comment (Darkoberd00) [reply src/commands/general/setParser.ts]: I have changed it so that it looks like this:

image

**Review by Schmarsch — APPROVED** [review]

**Pull request merged**